cmd/fdo_client/client.go

```diff
@@ -386,6 +396,29 @@ func di() (err error) { //nolint:gocyclo
 	return err
 }
 
+func getDeviceInfo() (string, error) {
+	if diDeviceInfo == DeviceInfoSerialNumber {
+		productSerial, err := os.ReadFile("/sys/devices/virtual/dmi/id/product_serial")
+		if err == nil && len(productSerial) != 0 {
+			return strings.TrimSpace(string(productSerial)), nil
+		}
+		chassisSerial, err := os.ReadFile("/sys/devices/virtual/dmi/id/chassis_serial")
+		if err == nil && len(chassisSerial) != 0 {
+			return strings.TrimSpace(string(chassisSerial)), nil
+		}
+		slog.Warn("error determining serial number for device")
+		return "", nil
+	}
```

Comment:
what I can add is the logic to fallback to the very first available network card's mac address if serial is empty
let me know